### ISSUE TITLE
Confirm #155

### ISSUE THREAD

**Issue by jGleitz** [jGleitz/JUnit-KIT#166]
#155 introduced a move scenario some other tests rely on. Plus, it tests many essential parts of the ant's movement. Sombody has to confirm that **all Javadoc** and all tests are correct.

**Comment by pcworld** [comment]
Using a8ef9f4b8de87a21a65596b2cc49342190353c9e, I have verified that:
- MoveTest#testConsistency's checked pitches are correct
- MoveTest#allTypesMoveTest is correct, ALL_TYPES_PITCHES is correct and have verified the JavaDoc of ALL_TYPES_BOARD; my findings are:  

``` java
    /**
     * A test file containing all types of ants, colours and cells. Let's take a look at what we expect to happen:
     * <p>
     * We start with this board:
     * 
     * 
     * <pre>
     * print|colour|direction
     * 
     * abir | 0000 | ↓↓↓↓
     * 0041 | 0041 |     
     * 0*00 | 0*00 |     
     * CDJS | 0000 | ↑↑↑↑ // TODO "cdjs" should be lowercase
     * </pre>
     * 
     * We then run {@code move 1}. This should first move the standard ants. The athletic ants are next. {@code speedup}
     * is {@code 4}, {@code rule} is {@code 90-90-315-90-270} so they perform four steps. Last, the lazy ants move. We
     * get the last figure as first result:
     * 
     * <pre>
     * 00ir | 0000 |   ↓↓ 
     * ab41 | 3341 | ←←  
     * c*00 | 3*00 | →   
     * 0djs | 0300 |  →↑↑
     * ------------------ // move of j tests "collision of ants" part of #141; tests that an athletic ant can leave
     * 000r | 0000 |    ↓ //                                                   the field in between its 4 usual moves
     * ab42 | 3342 | ←←   
     * c*i3 | 3*33 | → ↑  
     * 0d0s | 0300 |  → ↑ 
     * ------------------
     * 0000 | 0000 |     
     * ab4r | 3341 | ←← ↘
     * c*is | 3*30 | → ↑→
     * 0d00 | 0300 |  →
     * </pre>
     * 
     * Next {@code move 1}:
     * 
     * <pre>
     * 0000 | 0000 |    
     * b34r | 0341 | ↑  ↘  // tests "that the order of their moves is valid" part of #141 to a certain degree; see text below this code snippet
     * c*is | 0*30 | ↓ ↑→
     * 03d0 | 0330 |   ↓ 
     * ------------------
     * 03i0 | 0330 |   ↓      
     * b04r | 0041 | ↑  ↘ 
     * c*3s | 0*30 | ↓  → 
     * 03d0 | 0330 |   ↓  
     * ------------------
     * 03i0 | 0330 |   ↓  
     * b04r | 0041 | ↑  ↘ 
     * c*3s | 0*30 | ↓  → 
     * 03d0 | 0330 |   ↓
     * </pre>
     * 
     * Another {@code move 1}:
     * 
     * <pre>
     * b3i0 | 3330 | → ↓  // move of d tests #138: "Ants that have left the field are no obstacle"  
     * 004r | 0041 |    ↘ 
     * 0*3s | 0*00 |    → // bug: color row should be "0*30"
     * c330 | 3330 | ←    
     * ------------------
     * b30i | 3303 | →  ↓  
     * 004r | 0041 |    ↘ 
     * 0*3s | 0*00 |    → // bug: color row should be "0*30"
     * c330 | 3330 | ←    
     * ------------------
     * b30i | 3303 | →  ↓  
     * 004r | 0041 |    ↘ 
     * 0*3s | 0*00 |    → // bug: color row should be "0*30"
     * c330 | 3330 | ←
     * </pre>
     * 
     * Another {@code move 1}:
     * 
     * <pre>
     * 3b0i | 3003 |  ↓ ↓
     * 004r | 0041 |    ↘
     * 0*3s | 0*00 |    → // bug: color row should be "0*30"
     * 3330 | 3330 |     
     * ------------------
     * 3b30 | 3030 |  ↓    
     * 004r | 0041 |    ↘  
     * 0*3s | 0*00 |    → // bug: color row should be "0*30"
     * 3330 | 3330 |       
     * ------------------
     * 3b30 | 3030 |  ↓   
     * 004r | 0041 |    ↘ 
     * 0*3s | 0*00 |    → // bug: color row should be "0*30"
     * 3330 | 3330 |
     * </pre>
     * 
     * Another {@code move 1}:
     * 
     * <pre>
     * 3030 | 3030 |      
     * 0b4r | 0341 |  ← ↘ 
     * 0*3s | 0*00 |    → // bug: color row should be "0*30"
     * 3330 | 3330 |
     * ------------------
     * 3030 | 3030 |      
     * 0b4r | 0341 |  ← ↘ 
     * 0*3s | 0*00 |    → // bug: color row should be "0*30"
     * 3330 | 3330 |      
     * ------------------
     * 3030 | 3030 |      
     * 0b41 | 0341 |  ←   
     * 0*30 | 0*00 |      // bug: color row should be "0*30"
     * 3330 | 3330 |
     * </pre>
     */
```

Re the claim from the original pull request:

> Adds a more complex move test, containing all types of cells, colours and ants

While it is true that all colors are at some point found on the field, not all colors are contained in the tested pitches; while color `2` does not appear in any of the checked pitches, it is still tested to a certain degree since the color mapping of `2` to 315° results in `r`'s departure from the field in the last move.
The test assures that obstacles are correctly recognized and that the order of (standard) ants' moves is correct (at least in one case of `a` and `b`) as they would collide otherwise; however `a` and `b` is the order of ants on the original field and thus it **does not** assure that the order of the moves is alphabetical rather than the order of the ants' original appearance. The only test that tests this (incidentally?) is some test(s?) in [PositionTest#moveAntPositionTest](https://github.com/jGleitz/JUnit-KIT/blob/a8ef9f4b8de87a21a65596b2cc49342190353c9e/src/final2/subtests/PositionTest.java#L122); I've created #173 for this.
- MoveTest#terminateAfterAllHaveLeftTest is correct and tests all three `move` cases of #142.  
  I suggest to add a test with `speedup=10` (or higher) to make sure the program exits cleanly even if there is one (or more) move(s) "left".
- MoveTest#noZombieAntsTest correctly fixes #138.

Edit: I have **not** reviewed #170.

**Comment by jGleitz** [comment]
Thanks a lot. I think that's all that's needed.
